core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/OpenAPIGenerator.java

```diff
@@ -29,15 +32,45 @@ public static synchronized OpenAPIGenerator getInstance() {
         return instance;
     }
 
-    public void generate(List<Connection> connections, List<Component> components, String target, boolean load) {
+    public TransformResult transformFromFile(String path) {
+        File docFile = new File(path);
+        SwaggerParseResult result = new OpenAPIParser().readLocation(String.valueOf(docFile), null, null);
+        checkForMessages(result);
+
+        return new TransformResult(
+                ConnectionParser.getInstance().parse(result.getOpenAPI()),
+                ComponentParser.getInstance().parse(result.getOpenAPI())
+        );
+    }
+
+    public TransformResult transformFromJsonContent(String doc) {
+        SwaggerParseResult result = new OpenAPIParser().readContents(doc, null, null);
+        checkForMessages(result);
+
+        return new TransformResult(
+                ConnectionParser.getInstance().parse(result.getOpenAPI()),
+                ComponentParser.getInstance().parse(result.getOpenAPI())
+        );
+    }
+
+
+    private void checkForMessages(SwaggerParseResult result) {
+        if (result.getMessages() != null) {
+            result.getMessages().forEach(log::warn);
+        }
+    }
+
+    public void generate(TransformResult transformResult, String target, boolean load) {
+        Configuration.getInstance();
+        FrameworkCrypto.getInstance();
```

Comment:
This should be done at start up of the application. When starting the REST server its already done in the Bean FrameworkInstance. When running CLI operations like iesi-metadata.sh, iesi-openapi.sh it should be done in the linked Launcher